### PR TITLE
Refactor/ben/click

### PR DESCRIPTION
**Pull request by ben-bay**
Here's the click refactor (doing this first to apply to internal spellbook later).

I wish we had unit tests and cli tests to be sure I didn't break anything

### CODE REVIEW

File: spellbook/commands/serialize.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    required=False,
+    default="output.json",
+    type=click.File("wb"),
+    help="output file",
+)
+@click.option(
+    "--vars",
+    required=False,
+    default="results.hdf5",
+    type=str, # TODO nargs="+"
```
Comment (lucpeterson) [inline spellbook/commands/serialize.py:18]: todo

File: spellbook/commands/stack-npz.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+@click.argument(
+    "target",
+    required=True,
+    type=click.File("wb"),
+    #help="target file",
+)
+@click.argument(
+    "source",
+    required=True,
+    type=str,#TODO nargs="+"
```
Comment (lucpeterson) [inline spellbook/commands/stack-npz.py:25]: todo

**Review by lucpeterson — CHANGES_REQUESTED** [review]
I think we need to go through a lot of tests to see how much this breaks. Not sure how to do it, besides manually.

**Comment by lucpeterson** [comment]
Since the old arg parser is there in each command, could we just compare them, running, eg
python stack_npz.py my_args
spellbook stack_npz my_args
to see if they're the same?

**Comment by ben-bay** [comment]
@lucpeterson `make-samples` works the same as the original. I don't have time today to test the others, but I used the same methods across all 7 commands.

File: spellbook/data_formatting/conduit/python/translator.py [hunk 3]
```diff
@@ -29,42 +29,10 @@
...
 
-import argparse
 import sys
 
 import numpy as np
 
-from spellbook.utils import prep_argparse
-
-
-def import_conduit():
```
Comment (lucpeterson) [inline spellbook/data_formatting/conduit/python/translator.py:40]: we probably need to keep conduit

File: spellbook/data_formatting/serialize.py [hunk 4]
```diff
@@ -36,58 +33,3 @@ def nested_dict(var_list, splitter="/"):
         value = maybe_numeric_or_bool(val)
         nested_set(output, keylist, value)
     return output
-
-
-def process_args(args):
```
Comment (lucpeterson) [inline spellbook/data_formatting/serialize.py:41]: probably need to keep this function

**Review by lucpeterson — CHANGES_REQUESTED** [review]
check that these all run. i think some bits of code were unintentionally deleted

**Pull request merged**